packages/components/src/radio-group/index.tsx

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import * as Ariakit from '@ariakit/react';
+
+/**
+ * WordPress dependencies
+ */
+import { useMemo, forwardRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import ButtonGroup from '../button-group';
+import type { WordPressComponentProps } from '../context';
+import { RadioGroupContext } from './context';
+import type { RadioGroupProps } from './types';
+
+function UnforwardedRadioGroup(
+	{
+		label,
+		checked,
+		defaultChecked,
+		disabled,
+		onChange,
+		children,
+		...props
+	}: WordPressComponentProps< RadioGroupProps, 'div', false >,
+	ref: React.ForwardedRef< any >
+) {
+	const radioStore = Ariakit.useRadioStore( {
+		value: checked,
+		defaultValue: defaultChecked,
+		setValue: ( newValue ) => {
+			onChange?.( newValue ?? undefined );
+		},
+	} );
+
+	const contextValue = useMemo(
+		() => ( {
+			store: radioStore,
+			disabled,
+		} ),
+		[ radioStore, disabled ]
+	);
+
+	return (
+		<RadioGroupContext.Provider value={ contextValue }>
+			<Ariakit.RadioGroup
+				store={ radioStore }
+				render={ <ButtonGroup>{ children }</ButtonGroup> }
```

Comment:
In principle, I'd prefer if we passed `children` to `Ariakit.RadioGroup` — would TypeScript be ok with that?

Reply:
If we don't pass on `children` like this

```diff
- render={ <ButtonGroup>{ children }</ButtonGroup> }
+ render={ <ButtonGroup /> }
```

and have spread it via `{ ...props }` on `Ariakit.RadioGroup` then TypeScript complains with the following error message:

```
Property 'children' is missing in type '{}' but required in type 'ButtonGroupProps'. ts(2741)
```

I'm not sure why that is but we wondered whether @diegohaz might have any suggestions on what's the preferred way to pass on `children` in a case like this.

Reply:
It appears that `ButtonGroup` requires the `children` prop to be explicitly passed. If that's the case, you should either pass it or make it optional in `ButtonGroupProps`.

Reply:
Thank you — @flootr  I guess we can keep the code as-is (we're already passing `children` explicitly)